src/common.rs

```diff
@@ -415,3 +415,10 @@ pub enum SymbolFlags<Section> {
         associative_section: Option<Section>,
     },
 }
+
+/// Unified segment rwe_flag of read permission
+pub const SEGMENT_RWE_FLAG_READ: u32 = 0x01;
+/// Unified segment rwe_flag of write permission
+pub const SEGMENT_RWE_FLAG_WRITE: u32 = 0x02;
+/// Unified segment rwe_flag of execute permission
+pub const SEGMENT_RWE_FLAG_EXECUTE: u32 = 0x04;
```

Comment:
Instead of these, can you add an `enum SegmentFlags` (similar to existing `SectionFlags`), and give it methods for `is_readable` etc.

For Mach-O, we can include both `initprot` and `maxprot` in the flags (but the methods will only use `initprot`).